### PR TITLE
GDB-5341 migrate to Jenkins

### PR DESCRIPTION
**Pull request by svilenvelikov**
Migrate the CI process from TravisCI to Jenkins.

* Add Jenkinsfile for MR and release
* Removed travis.yaml config
* Configured sonarcloud analisys and removed codeclimate and coveralls

Steps after this is merged:
* change the PR job to point to current branch and not to this specific one
* detach github repo from travis
* detach github repo from codeclimate
* switch the docs from gh-pages branch to master and /docs folder

### CODE REVIEW

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Comment by svilenvelikov** [comment]
@mihailradkov would you have the time to review this MR? I can't assign you as reviewer for some reason.

**Comment by teodossidossev** [comment]
Will you consider archiving ```.travis.yml``` file instead of deleting it?

**Comment by svilenvelikov** [comment]
@teodossidossev 
> Will you consider archiving .travis.yml file instead of deleting it?

Why do you need that? You can always get it from git.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  environment {
+    CI = "true"
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        sh "node --version"
```
Comment (desislava-hristova-ontotext) [inline Jenkinsfile:14]: Was this for debug or there is a other purpose for it?

File: JenkinsfileRelease [hunk 2]
```diff
@@ -0,0 +1,85 @@
+pipeline {
+
+  parameters {
```
Comment (desislava-hristova-ontotext) [inline JenkinsfileRelease:3]: Maybe we should at some time think of extracting and reusing the common stuff between the projects.

**Review by desislava-hristova-ontotext — DISMISSED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (svilenvelikov) [reply Jenkinsfile:14]: It has informative purpose only becase we don't know what's the current version of the nodejs which is used.

File: JenkinsfileRelease (hunk 2 above)
Comment (svilenvelikov) [reply JenkinsfileRelease:3]: Yes. @mihailradkov mentioned before that he plans to make the utility functions library currently used for the platform more generic. Then we can use it too.

**Review by teodossidossev — APPROVED** [review]

**Review by desislava-hristova-ontotext — APPROVED** [review]

**Review by desislava-hristova-ontotext — APPROVED** [review]

**Pull request merged**